include/cuco/detail/static_set/kernels.cuh

```diff
@@ -89,6 +90,193 @@ CUCO_KERNEL void find(InputIt first, cuco::detail::index_type n, OutputIt output
+  // random choice to tune
+  auto constexpr flushing_buffer_size = 2 * flushing_tile_size;
+  auto constexpr num_flushing_tiles   = BlockSize / flushing_tile_size;
+  auto constexpr max_matches          = flushing_tile_size / tile_size;
+
+  auto const flushing_tile    = cg::tiled_partition<flushing_tile_size>(block);
+  auto const flushing_tile_id = block.thread_rank() / flushing_tile_size;
+
+  __shared__ cuco::pair<size_type, size_type>
+    flushing_tile_buffer[num_flushing_tiles][flushing_tile_size];
+  // per flushing-tile counter to track number of filled elements
+  __shared__ size_type flushing_counter[num_flushing_tiles];
+
+  if (flushing_tile.thread_rank() == 0) { flushing_counter[flushing_tile_id] = 0; }
+  flushing_tile.sync();  // sync still needed since cg.any doesn't imply a memory barrier
+
+  while (flushing_tile.any(idx < n)) {
+    bool active_flag = idx < n;
+    auto const active_flushing_tile =
+      cg::binary_partition<flushing_tile_size>(flushing_tile, active_flag);
+    if (active_flag) {
+      auto const found = hash_table.find(tile, *(iter + idx));
+      if (tile.thread_rank() == 0 and found != hash_table.end()) {
+        auto const offset = atomicAdd_block(&flushing_counter[flushing_tile_id], 1);
+        flushing_tile_buffer[flushing_tile_id][offset] = cuco::pair{
+          static_cast<size_type>(found->second), static_cast<size_type>(idx)};
+      }
+    }
+
+    flushing_tile.sync();
+    if (flushing_counter[flushing_tile_id] + max_matches > flushing_buffer_size) {
+      flush_buffer(flushing_tile,
+                   flushing_counter[flushing_tile_id],
+                   flushing_tile_buffer[flushing_tile_id],
+                   counter,
+                   build_indices,
+                   probe_indices);
+      flushing_tile.sync();
+      if (flushing_tile.thread_rank() == 0) { flushing_counter[flushing_tile_id] = 0; }
+      flushing_tile.sync();
+    }
+
+    idx += stride;
+  }  // while
+
+  if (flushing_counter[flushing_tile_id] > 0) {
+    flush_buffer(flushing_tile,
+                 flushing_counter[flushing_tile_id],
+                 flushing_tile_buffer[flushing_tile_id],
+                 counter,
+                 build_indices,
+                 probe_indices);
+  }
+  */
+}
+
+template <typename Size,
+          typename ProbeKey,
+          typename Key,
+          typename AtomicT,
+          typename OutputIt1,
+          typename OutputIt2>
+__device__ void flush_buffer(cooperative_groups::thread_block const& block,
+                             Size buffer_size,
+                             cuco::pair<ProbeKey, Key>* buffer,
+                             AtomicT* counter,
+                             OutputIt1 probe_begin,
+                             OutputIt2 output_begin)
+{
+  auto i = block.thread_rank();
+  __shared__ Size offset;
+
+  if (i == 0) { offset = counter->fetch_add(buffer_size, cuda::std::memory_order_relaxed); }
+  block.sync();
```

Comment:
*question*: Is this more future-proofly written as:

```
Size offset = cooperative_groups::invoke_one_broadcast(block, [&]() { return counter->fetch_add(buffer_size, cuda::std::memory_order_relaxed); });
```
?

Reply:
Good point.

I will do this in the tile-wise `flush_buffer`.